pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -127,10 +124,31 @@ func (ctrl *Controller) syncNodeConfigHandler(key string) error {
 		if err != nil {
 			return err
 		}
-		// updating the kubelet configuration with the Node specific configuration.
-		err = updateOriginalKubeConfigwithNodeConfig(nodeConfig, originalKubeConfig)
-		if err != nil {
-			return err
+		// the workerlatencyprofile's configuration change will be applied only on the worker nodes.
+		if role == ctrlcommon.MachineConfigPoolWorker {
+			// updating the kubelet configuration with the Node specific configuration.
+			err = updateOriginalKubeConfigwithNodeConfig(nodeConfig, originalKubeConfig)
+			if err != nil {
+				return err
+			}
+		}
+		if features.Spec.FeatureGateSelection.FeatureSet == osev1.TechPreviewNoUpgrade {
+			featureSet := osev1.FeatureSets[osev1.TechPreviewNoUpgrade]
+			for _, val := range featureSet.Enabled {
+				if val == "CGroupsV2" {
+					// updating the machine config resource with the relevant cgroup configuration
+					err := updateMachineConfigwithCgroup(nodeConfig, mc)
+					if err != nil {
+						return err
+					}
+					break
+				}
+			}
+		} else if nodeConfig.Spec.WorkerLatencyProfile == "" {
```

Comment:
sorta confused why we need to add WorkerLatency Profile here?

Reply:
Workerlatency profile gets applied only on the worker pool whereas the cgroup config is for every pool. As both of these fields are in a single resource(config-node), we need a way to apply the changes to a specific pool. These checks help in preventing the creation of unnecessary machine config objects.